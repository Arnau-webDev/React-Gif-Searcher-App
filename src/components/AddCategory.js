import React, {useState} from 'react';
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories, categories }) => {

const [inputValue, setInputValue] = useState("");

const handleInputChange = (e) => {
    setInputValue(e.target.value);
}

const handleSubmit = (e) => {
    e.preventDefault();

    categories.forEach((element, index) => {
        if(element.toUpperCase() === inputValue.toUpperCase()) {
            categories.splice(index, 1);
        }
    })
    
    if(inputValue.trim().length > 1 && inputValue.trim().length < 35 ) {
        setCategories(( categories ) => [inputValue, ...categories]);
        setInputValue("");
    }

    setInputValue("");
}

    return (
        <form onSubmit={ handleSubmit }>
            <input
                type="text"
                placeholder="Type about something..."
                value= { inputValue }
                onChange= { handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}
