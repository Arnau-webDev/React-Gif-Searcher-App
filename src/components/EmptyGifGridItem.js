import React from 'react'

export const EmptyGifGridItem = ({ category }) => {

    return (
        <div className="notFoundCard animate__animated animate__fadeIn">
            <p>No GIFS where found from the <span>{ category }</span> search</p>
        </div>
    )
}
