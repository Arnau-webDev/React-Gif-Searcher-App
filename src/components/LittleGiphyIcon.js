import React from 'react';

export const LittleGiphyIcon = () => {

    const littleIconUrl = "https://media0.giphy.com/media/3xz2BDFvxop2BfAQoM/100w.gif?cid=6f54fedeoerrqrxeblu5mj93k2ryzyd366dbj9sca30ajebk&rid=100w.gif";

    return (
        <div>
            <div className="smallContainer">
                <div className="small animate__animated animate__backInDown">
                    <a href="https://developers.giphy.com/" alt="" target="_blank" rel="noopener noreferrer">
                        <div>Powered by GIPHY</div>
                        <div className="extraSmall"><img src={ littleIconUrl } alt=""></img></div>
                    </a>
                </div>
            </div>
            <h2>Gif Expert App</h2>
        </div>
    )
}
