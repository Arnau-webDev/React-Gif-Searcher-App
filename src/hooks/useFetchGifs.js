import { useState, useEffect } from 'react';
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = ( category ) => {

    const [state, setState] = useState({
        data: [],
        loading: true,
        dataLength: 0,
        categoriesSubmitted: [],
    });

    useEffect( () => {
        getGifs (category)
            .then((imgs) => { setTimeout(() => {
                setState({
                    data: imgs,
                    loading: false,
                    dataLength: imgs.length
                });
            }, 550)});
    }, [ category ]);

    return state;
}
