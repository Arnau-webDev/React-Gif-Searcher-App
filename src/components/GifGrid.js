import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import { EmptyGifGridItem } from './EmptyGifGridItem';

export const GifGrid = ({ category }) => {

    // const categoriesAlreadySearched = document.querySelectorAll("h3");

    //  categoriesAlreadySearched.forEach((element) => {
    //      console.log(element.innerHTML)
    //     if(element.innerHTML.toUpperCase() === category.trim().toUpperCase() ){
    //         return <></>
    //     }
        
    //  }); 

    const { data:images, loading, dataLength } = useFetchGifs( category );

    return (
        <div>
            <h3 className="animate__animated animate__fadeIn">{ category }</h3>
            
            { loading && <p className="animate__animated animate__flash">Loading...</p> }
            { dataLength < 1 && loading === false && <EmptyGifGridItem category={ category } /> }

            <div className="card-grid">
                { images.map(( img ) => <GifGridItem key={ img.id } img={ img } /> )}
            </div>
        </div>
    )
}
