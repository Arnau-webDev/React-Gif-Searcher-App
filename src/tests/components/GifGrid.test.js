import React from 'react';
import '@testing-library/jest-dom';

import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
const { shallow } = require('enzyme');

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas <GifGrid />', () => {
	const category = 'Goku';

	test('Debe mostrar el contenido correctamente', () => {
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true,
			dataLength: 0,
		});

		const wrapper = shallow(<GifGrid category={category} />);

		expect(wrapper).toMatchSnapshot();
	});

	test('Debe de mostrar los items despues de la consulta fetch', () => {
		const gifs = [
			{
				id: 'ABC',
				title: 'Some title',
				url: 'https://localhost/myImgae.png',
			},
		];

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
			dataLength: gifs.length,
		});

		const wrapper = shallow(<GifGrid category={category} />);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('p').exists()).toBe(false);
		expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
	});
});
