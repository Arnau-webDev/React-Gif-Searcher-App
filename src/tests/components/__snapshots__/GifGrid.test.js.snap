// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifGrid /> Debe de mostrar los items despues de la consulta fetch 1`] = `
<div>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      img={
        Object {
          "id": "ABC",
          "title": "Some title",
          "url": "https://localhost/myImgae.png",
        }
      }
      key="ABC"
    />
  </div>
</div>
`;

exports[`Pruebas <GifGrid /> Debe mostrar el contenido correctamente 1`] = `
<div>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</div>
`;
